### PR TITLE
NEW Install dynamodb if required

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-dynamodb/issues/47

Tag 1.2.3 when merged

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -193,11 +193,14 @@ jobs:
           fi
           echo "PHP has been configured"
 
-      - name: Apt install additional requirements 
+      - name: Install additional requirements 
         env:
           GITHUB_REPOSITORY: ${{ github.repository }}
+          # These are used for aws-cli, which is pre-installed in github actions. This is only for dynamodb
```
Comment (ChloeSartorelli) [inline .github/workflows/ci.yml]: Does this mean these are required for dynamodb installation to work?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -209,6 +212,22 @@ jobs:
...
             sudo apt install -y hunspell libhunspell-dev hunspell-en-us
           fi
+          if [[ $GITHUB_REPOSITORY =~ /(silverstripe-dynamodb)$ ]]; then
+            # install dynamodb
+            mkdir _dynamodb
+            cd _dynamodb
+            sudo wget https://s3.ap-southeast-1.amazonaws.com/dynamodb-local-singapore/dynamodb_local_latest.tar.gz
+            sudo tar -xf dynamodb_local_latest.tar.gz
+            # run dynamo db as background process
+            # github actions comes java - `java -version` = openjdk version "11.0.18" 2023-01-17
```
Comment (ChloeSartorelli) [inline .github/workflows/ci.yml]: What does "github actions comes java" mean and why did you include a java command in this comment?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -209,6 +212,22 @@ jobs:
...
+            $(java -Djava.library.path=./DynamoDBLocal_lib -jar DynamoDBLocal.jar) &
+            sleep 3
+            if ! [[ $(ps -aux | grep DynamoDBLocal.jar) =~ DynamoDBLocal\.jar ]]; then
```
Comment (ChloeSartorelli) [inline .github/workflows/ci.yml]: What is the `sleep 3` there for?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -570,6 +590,16 @@ jobs:
           SS_BASE_URL="http://localhost"
           EOF
 
+          if [[ $GITHUB_REPOSITORY =~ /(silverstripe-dynamodb)$ ]]; then
+            cat << EOF >> .env
+          AWS_DYNAMODB_ENDPOINT="http://localhost:8000"
```
Comment (ChloeSartorelli) [inline .github/workflows/ci.yml:599]: Should this stuff be indented to match the indentation of `cat`?

File: .github/workflows/ci.yml (hunk 3 above)
Comment (emteknetnz) [reply .github/workflows/ci.yml]: Ensure the that DynamoDBLocal.jar has started, it's running as a background process so it goes to the next line without waiting for the command to finish.  In hindsight I should change this to a for loop and check to that it's started

File: .github/workflows/ci.yml (hunk 4 above)
Comment (emteknetnz) [reply .github/workflows/ci.yml:599]: No, it's deliberate, this indentation is correct. We do the same thing higher up in the file.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (emteknetnz) [reply .github/workflows/ci.yml]: Actually we can probably get rid of these now that we're not doing the `aws dynamodb` commands in bash anymore

File: .github/workflows/ci.yml (hunk 2 above)
Comment (emteknetnz) [reply .github/workflows/ci.yml]: Means java is installed out of the box, it doesn't seem it's a vanilla install of ubuntu which lacks java (at least on my local it did).  Will update comment

**Review by GuySartorelli — APPROVED** [review]
Looks good. The logic all works locally.

**Comment by GuySartorelli** [comment]
Tagged as [1.2.3](https://github.com/silverstripe/gha-ci/releases/tag/1.2.3)

**Pull request merged**